lib/include/prjxray/xilinx/xc7series/ecc.h

```diff
@@ -0,0 +1,50 @@
+#ifndef PRJXRAY_LIB_XILINX_XC7SERIES_ECC_H_
+#define PRJXRAY_LIB_XILINX_XC7SERIES_ECC_H_
+
+#include <cstdint>
+
+namespace prjxray {
+namespace xilinx {
+namespace xc7series {
+
+// Extend the current ECC code with one data word (32 bit) at a given
+// index in the configuration packet and return the new ECC code.
```

Comment:
I don't understand why index is defined relative to the configuration packet.  I thought the ECC was defined only in terms of a single frame.  Normal bitstreams use a single configuration packet to write the whole configuration.  Won't that wreak havoc on this method?